src/readii/feature_extraction.py

```diff
@@ -318,6 +322,9 @@ def featureExtraction(ctSeriesID):
             delayed(featureExtraction)(ctSeriesID) for ctSeriesID in ctSeriesIDList
         )
 
+    # Filter out None and ensure each result is a list (even if it's empty)
+    features = [f if isinstance(f, list) else [f] for f in features if f is not None]
```

Comment:
What is this doing? features should be a list of dictionaries, does this maintain that?

Reply:
yeah the Parallel route returns a Generator instead of a list so this ensures that its a list since thats what `chain.from_iterable` expects

Reply:
also theres no assertion anywhere to make sure that its a list of dictionaries, since the nested `featureExtraction` function doesnt really indicate what it returns.

at a glance it looks like it returns a list that might hold `dicts` 

perhaps it could use some type annotations to clarify that it returns a list of dictionaries or None (if the list is empty) 

though I dont think this is a bug or something that might break anything... 

wdyt?

Reply:
I agree that it needs to be clarified for sure.
However, I'm not sure the parallelization is currently being used anywhere. It made it super hard for us to debug stuff on H4H and then it's not needed in the snakemake pipeline. 
So I'm wondering if it's even worth keeping?